code/modules/mining/machine_processing.dm

```diff
@@ -156,6 +169,89 @@
 	src.updateUsrDialog()
 	return
 
+/obj/machinery/mineral/processing_unit_console/proc/print_report(var/mob/living/user)
```

Comment:
If you make this proc sleep with CHECK_TICK (or was it TICK_CHECK), then add a busy flag. To stop the initiation of new prints while this one is running.